WTF is a forward deployed engineer? (#14825)
PostHog/posthog.com: contents/blog/forward-deployed-engineer.md

## Why does it feel like everyone is hiring forward deployed engineers right now?

The obvious answer is AI. 

Since AI is a completely new type of product, there are too many problems that have never been solved before. Talking to users helps engineers identify problems to build for. But that's what product engineers already do – why are people taking a different approach now?

It's because AI requires lots of custom-fitting to work properly. [Anjor Kanekar](https://anjor.xyz/), an FDE at Palantir for more than seven years, puts it [this way](https://newsletter.pragmaticengineer.com/p/forward-deployed-engineers):

"AI companies have a problem that the FDE is really well suited to solve. There is a gap between foundational model capabilities and the application of those in enterprise use cases where they can add value. The FDE model is a way of doing the R&D work of figuring out the 'app layer' in a really effective way. AI companies have the models, but they need to figure out what exactly needs to be built using the models to make inroads into large enterprises / public sector."

## What kinds of people become forward deployed engineers?

> Companies look to [hire](https://builders.ramp.com/post/forward-deployed-engineering#:~:text=to%20learn%20from.-,How%20we%20hire%20FDEs,-In%20order%20to) forward deployed engineer is someone with...

Reviewer: Any PostHog points to add?

[Shyam Sankar](https://www.linkedin.com/in/shyamsankar), President and CTO of Palantir, has been [quoted](https://www.youtube.com/watch?v=Zyw-YA0k3xo) saying that people who are “heretics” or “rebels” in their domain make for perfect FDEs because they have the unique depth, context, and energy that can unlock 3x-10x growth for a business.

If you haven't noticed yet, all of these are traits that also make for great product engineers and founders. It's no wonder why so many former Palantir employees go on to [start their own companies](https://www.businessinsider.com/palantirs-forward-deployed-engineer-role-churns-out-startup-founders-2025-6). 

## When should I hire our first forward deployed engineer?

If the product-market fit playbook is already working for you, the answer is that [you shouldn’t](https://www.youtube.com/watch?v=Zyw-YA0k3xo). 

(Image: dont try this at home kids meme)

In the [FDE model](https://www.futureventures.ca/insights/understanding-the-forward-deployed-engineering-model), you lose a LOT of money early on. 

> You start by solving one of their CEO’s top five problems, then you earn the right to solve bigger problems for them. But until that happens, you are spending lots of money and headcount to keep engineers onsite. This can take multiple years for a single customer.

Reviewer 2: This is where I feel we could consider talking about the first FDE at PostHog will be remote first, we're challenging the trend and cost implications by raising that we already have a lot of data and a developed async culture that we want to leverage to discover our own brand of FDE. It's a weird experiment for those who are used to the standard FDE spiel, but it makes sense when you think about it, many of the benefits of FDE could still be leveraged for way less money, and that's what we're exploring.

## How are they different from software engineers?

FDEs still do software engineering, but they spend way more time [talking to customers](/newsletter/talk-to-users) and scoping projects [than other software engineers](https://blog.palantir.com/dev-versus-delta-demystifying-engineering-roles-at-palantir-ad44c2a6e87) to decide _what_ to build, and how. Their top priority is to deeply understand a customer's problems so that they can develop something that addresses it.

Software engineers on the main product team, on the other hand, focus on building features that are generalizable and maintainable. They're less involved in direct customer conversations and more focused on solutions that work at scale.

[Bob McGrew](https://www.youtube.com/watch?v=Zyw-YA0k3xo), an early executive at Palantir, describes FDEs as the ones who build the rough gravel road for where the product needs to go. Software engineers from the core product team then work on turning that gravel road into a paved superhighway so that the next ten customers can also use it.

> The key difference is that FDEs work on product discovery and development with _one_ customer at a time, and generalizability is a second priority. Software engineers typically build with many users in mind from the start, and thinking about scale is baked into their process.

Reviewer: I think you could turn these three paragraphs into two, you are basically repeating yourself a bit here.

- Software engineer is generalizable and thinking of scale from the start.
- FDE is one customer at a time
- Gravel road vs paved road.

**3. You're targeting new verticals or segments**

When your company is exploring entirely new customer profiles and ICPs, you don't know what you don't know. FDEs can embed with early customers to understand the space and do all the product discovery for breaking into that market. 

This is what happened with Ramp, a team that originally started out building for [small businesses](https://ramp.com/blog/expense-tracker-for-business) but added FDEs as they ran into technical challenges trying to meet the [needs of enterprise](https://builders.ramp.com/post/forward-deployed-engineering).

## Why did we just hire a forward deployed engineer at PostHog?

We pride ourselves on having tools that work out of the box, but there's a difference between "any developer can use PostHog in 10 minutes" and "this Fortune 500 company can migrate their entire analytics stack without spending months on it."

> We hired our first FDE to help with these large-scale migrations, but we're doing it PostHog-style. Our FDE is fully remote, async, and focused on building automation. It's a weird experiment for those used to the typical FDE setup, but it makes sense when you think about it. Many of the benefits (deep customer knowledge, faster iterations, etc.) can still work if we use the tons of usage data we already have instead of doing product discovery from scratch.

Reviewer: ```suggestion
We hired our first FDE to help with these large-scale migrations, but we're doing it PostHog-style. Our FDE is fully remote, async, and focused on building automation. It's a weird experiment for those used to the typical FDE setup, but it makes sense for us. Many of the benefits (deep customer knowledge, faster iterations, etc.) can still work if we use the tons of usage data we already have instead of doing product discovery from scratch.
```
